### ISSUE TITLE
Problems using obfuscated dll on Windows Phone 8.1

### ISSUE THREAD

**Issue by avanpo** [yck1509/ConfuserEx#318]
I'm obfuscating a PCL (targeting Windows 8.1 and Windows Phone 8.1) that's referenced by a Windows Store app (Windows Phone 8.1). I'm using Visual Studio 2013 to deploy the WP8.1 app (as Release) to a device (the emulator does not appear to want to run it at all). Half of the protections do not work, even some that are included in the normal preset. The ones that do appear to work include "anti debug", "anti ildasm", "constants", "ctrl flow", and "rename".

For example, using the following .crproj file:

```xml
<project outputDir="C:\...\myPCL\bin\Release\Confused" baseDir="C:\...\myPCL\bin\Release" xmlns="http://confuser.codeplex.com">
  <rule pattern="true" preset="none">
    <protection id="resources" />
  </rule>
  <module path="myPCL.dll" />
  <probePath>C:\Program Files (x86)\Windows Phone Kits\8.1\References\CommonConfiguration\Neutral</probePath>
</project>
```

(Note: That probepath is required to get ConfuserEx to run). When running the WP8.1 project, it won't run properly, with no functionality loading (although it doesn't crash).

Anyway, I get the following:
`A first chance exception of type 'System.MethodAccessException' occurred in myPCL.dll
A first chance exception of type 'System.TypeInitializationException' occurred in mscorlib.ni.dll`

Breaking on the exception didn't give me much besides a mangled message with some messed up unicode and reversing of the characters that roughly translated to the following:

`System.MethodAccessException: Attempt by method '<Module>. method access to '()'System.Reflection.Assembly.Load(Byte[])' failed.`

The Data property was also set to `System.Collections.ListDictionaryInternal`.

As another example, "ref proxy" instead of "resources" gives me the following output:
`A first chance exception of type 'System.IO.FileNotFoundException' occurred in myPCL.ni.dll
A first chance exception of type 'System.IO.FileNotFoundException' occurred in mscorlib.ni.dll
A first chance exception of type 'System.IO.FileNotFoundException' occurred in mscorlib.ni.dll`

The message "Could not load file or assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies. The system cannot find the file specified." is given.

**Comment by yck1509** [comment]
Hi,
Windows Store apps have lots of limitation on the usable methods, so some protections may not work.

**Comment by uilquegb** [comment]
Try to add another `<probePath>`:
```xml
<project outputDir="C:\...\myPCL\bin\Release\Confused" baseDir="C:\...\myPCL\bin\Release" xmlns="http://confuser.codeplex.com">
  <rule pattern="true" preset="none">
    <protection id="resources" />
  </rule>
  <module path="myPCL.dll" />
  <probePath>C:\Program Files (x86)\Windows Phone Kits\8.1\References\CommonConfiguration\Neutral</probePath>
  <probePath>C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\WindowsPhoneApp\v8.1</probePath>
</project>
```
I hope it helps you, please, let me know if it did.

**Comment by ramanan-java** [comment]
Hi I tried using `<probPath>` as you specified in my Xamarin.Froms Windows Phone 8.1 WinRT project.
In my project I am using one .dll file which is obfuscated externally. But it always stopped like above exception.
I have tested in Debug, Release and Store build generation.
And facing same exception in UWP application also.

Exception:
`System.TypeInitializationException {"The type initializer for '<Module>' threw an exception."}`

Same time its working fine in Android project.

For your reference added the exception attachment. 
Let me know is there any other things to do. Thanks in advance...

![wp81err](https://user-images.githubusercontent.com/5810504/27222522-2a10fde6-52aa-11e7-89c0-f68b1936a6eb.png)
